hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java

```diff
@@ -581,6 +581,13 @@ private int getWhenNoSSLDetected() {
     public static final HazelcastProperty MAX_WAIT_SECONDS_BEFORE_JOIN
             = new HazelcastProperty("hazelcast.max.wait.seconds.before.join", 20, SECONDS);
 
+    /**
+     * if true, join returns right away and background tasks are waiting for above timeouts
+     * this reduces latency when new members join clusters one-by-one
+     */
+    public static final HazelcastProperty WAIT_SECONDS_BEFORE_JOIN_ASYNC
```

Comment:
This property name is ambiguous to me - I understand it has been adapted from the existing properties, but the current name implies to me that the value is an integer (for amount of wait seconds when using an async strategy). I think we could rename it to something that more clearly shows what it does, i.e. `USE_ASYNC_MEMBER_JOIN_STRATEGY`

Reply:
I also think it might be a good idea to have this property default `false` in the final iteration so as to not impact existing functionality for other users who do not need speedy startup. (Keeping it true for now is good for test coverage though!)

Reply:
Agreed on the naming. Disagree on the default.

In general, my intent was to not have any sync mode at all, I just wound up having it as temporary compatibility sludge to be removed at a later point.

Sync code was never meant to be perfect or permanent, just temporary until some comfort has been developed with async strategy.

I consider the current sync method a bug and in an ideal world async method would be the one-and-only